ui/app/AppLayouts/Profile/panels/ContactsListPanel.qml

```diff
@@ -1,57 +1,163 @@
 import QtQuick 2.13
 import QtQuick.Controls 2.13
 import QtQuick.Layouts 1.13
+import QtQml.Models 2.13
 
 import utils 1.0
 import shared 1.0
 import shared.popups 1.0
+import shared.panels 1.0
 import "../../Chat/popups"
 
 import "."
 
-ListView {
-    id: contactList
+Item {
+    id: contactListRoot
+
     property var contactsModel
+    property int panelUsage: Constants.contactsPanelUsage.unknownPosition
 
-    property string searchStr: ""
+    property string title: ""
     property string searchString: ""
     property string lowerCaseSearchString: searchString.toLowerCase()
-    property string contactToRemove: ""
-    property bool hideBlocked: false
-    property bool showSendMessageButton
-
-    signal contactClicked(var contact)
-    signal openProfilePopup(var contact)
-    signal sendMessageActionTriggered(var contact)
-    signal openChangeNicknamePopup(var contact)
-
-    width: parent.width
-
-    model: contactList.contactsModel
-
-    delegate: ContactPanel {
-        id: panelDelegate
-        name: model.name
-        publicKey: model.pubKey
-        icon: model.icon
-        isIdenticon: model.isIdenticon
-        isContact: model.isContact
-        isBlocked: model.isBlocked
-        showSendMessageButton: contactList.showSendMessageButton
-
-        onClicked: contactList.contactClicked(model)
-        onOpenProfilePopup: contactList.openProfilePopup(model)
-        onSendMessageActionTriggered: contactList.sendMessageActionTriggered(model)
-        onOpenChangeNicknamePopup: contactList.openChangeNicknamePopup(model)
-
-        visible: {
-          if (hideBlocked && model.isBlocked) {
-            return false
-          }
-
-          return searchString === "" ||
-            panelDelegate.name.toLowerCase().includes(lowerCaseSearchString) ||
-            panelDelegate.publicKey.toLowerCase().includes(lowerCaseSearchString)
+
+    signal contactClicked(string publicKey)
+    signal openProfilePopup(string publicKey)
+    signal sendMessageActionTriggered(string publicKey)
+    signal openChangeNicknamePopup(string publicKey)
+    signal acceptContactRequest(string publicKey)
+    signal rejectContactRequest(string publicKey)
+    signal removeRejection(string publicKey)
+    signal textClicked(string publicKey)
```

Comment:
Signals should be in the past tense (end with 'ed') - this way it's easily distinguishable from function call + they should not assume what the receiver will do with them (`acceptContactRequest` -> `acceptContactClicked`)

Reply:
I'll just chime in to say that maybe this is better: `acceptContactRequest` -> `contactRequestAccepted`

Reply:
@osmaczko @jrainville yes, sounds good, signals naming updated.